File: src/PristaneLaverieSmart.Domain/Entities/MachineStatusAudit.cs
using PristaneLaverieSmart.Domain.Enums;

namespace PristaneLaverieSmart.Domain.Entities;

public class MachineStatusAudit
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid MachineId { get; set; }
    public MachineStatus OldStatus { get; set; }
    public MachineStatus NewStatus { get; set; }
    public DateTimeOffset OccurredOn { get; set; } = DateTimeOffset.UtcNow;
}

File: domain/src/main/java/ch/pristane/laverie/smart/domain/entities/MachineStatusAudit.java
package ch.pristane.laverie.smart.domain.entities;

import ch.pristane.laverie.smart.domain.enums.MachineStatus;

import java.time.OffsetDateTime;
import java.util.UUID;

public class MachineStatusAudit {
    private UUID id;
    private UUID machineId;
    private MachineStatus oldStatus;
    private MachineStatus newStatus;
    private OffsetDateTime occurredOn;

    public MachineStatusAudit() {
    }

    public MachineStatusAudit(UUID id, UUID machineId, MachineStatus oldStatus, MachineStatus newStatus, OffsetDateTime occurredOn) {
        this.id = id;
        this.machineId = machineId;
        this.oldStatus = oldStatus;
        this.newStatus = newStatus;
        this.occurredOn = occurredOn;
    }

    public static MachineStatusAudit create(UUID machineId, MachineStatus oldStatus, MachineStatus newStatus, OffsetDateTime occurredOn) {
        return new MachineStatusAudit(
                UUID.randomUUID(),
                machineId,
                oldStatus,
                newStatus,
                occurredOn
        );
    }

    public UUID getId() {
        return id;
    }

    public UUID getMachineId() {
        return machineId;
    }

    public MachineStatus getOldStatus() {
        return oldStatus;
    }

    public MachineStatus getNewStatus() {
        return newStatus;
    }

    public OffsetDateTime getOccurredOn() {
        return occurredOn;
    }
}